File: Problem4/problem4.c
#include <stdio.h>

int main()
{
	unsigned int num1 = 1000;
	unsigned int num2 = 999;
	unsigned int product = 0;
	unsigned int reverse_product = 0;
	unsigned int orig_product = 0;
	unsigned int max_palin = 0;

	while (num1-- > 1){
		num2 = 1000;
		while(num2-- > 0) {
			product = num1 * num2;
			orig_product = product;

			while (product != 0) {
				reverse_product = (reverse_product*10) + (product%10);
				product = product/10;
			}

			if(orig_product == reverse_product) {
				if(orig_product > max_palin)
					max_palin = orig_product;
			}

			reverse_product = 0;
			

		}
	}

	printf("Largest Palindrome Product = %u\n", max_palin);
}

File: Problem4/problem4.py
rev_number = 0
num1 = 999
num2 = 999
max_palin = 0;
num = 0

while num1 > 1:
        num2 = 999
        while num2 > 0:
                number = num1*num2
                num = number

                while number != 0:
                        rem = number%10
                        rev_number = rev_number*10 + rem
                        number = number//10


                if num == rev_number:
                        if num > max_palin:
                                max_palin = num

                num2 -= 1
                rev_number = 0
        num1 -= 1

print("Largest Palindrome Product = ", max_palin)
